### PR TITLE
WT-6765 Add more debugging and earlier detection of missing file.

### PR DESCRIPTION
**Pull request by sueloverso**

### CODE REVIEW

**Comment by sueloverso** [comment]
@ddanderson here's my current thoughts on new/more debugging for this rare failure. I originally added code to open the source directory with `O_DIRECT` but that is not allowed and always returns EINVAL. I could not find any documentation saying it is not allowed for directories, but it does not work (even with a hard-coded known pathname as a test). This has run several iterations on my test machine. Let me know if you have additional ideas.

**Comment by sueloverso** [comment]
@ddanderson I added the bool pointer because I didn't have the pid and `kill_child` could be static. Let me know if you prefer sending in the pid and making the kill function visible.

File: test/csuite/random_directio/util.c [hunk 1]
```diff
@@ -44,15 +44,15 @@
  *     Copy a directory, using direct IO if indicated.
  */
 void
-copy_directory(const char *fromdir, const char *todir, bool directio)
+copy_directory(const char *fromdir, const char *todir, bool directio, bool *fatal)
```
Comment (ddanderson) [inline test/csuite/random_directio/util.c:47]: Perhaps make `copy_directory` return `bool` rather than `void`?

**Review by ddanderson — APPROVED** [review]
LGTM, one possible suggestion to consider, but I'm good with either approach.

File: test/csuite/random_directio/util.c (hunk 1 above)
Comment (sueloverso) [reply test/csuite/random_directio/util.c:47]: Sure. I don't feel strongly either way but it doesn't pollute the arg list.

**Pull request merged**